contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -5,6 +5,16 @@ pragma solidity ^0.8.0;
 
 import "../ERC1967/ERC1967Proxy.sol";
 
+interface ITransparentUpgradeableProxy {
```

Comment:
We should add some docs explaining why this is defined here.

I don't like that we have to redefine the events here. They're not strictly necessary either but I can imagine it was necessary for the test suite to run. What do you think about definining an `IERC1967` interface with the events, and using that interface here and also in `ERC1967Upgrade`?